daffodil-io/src/main/scala/org/apache/daffodil/io/InputSource.scala

```diff
@@ -142,8 +145,9 @@ abstract class InputSource {
  *
  * @param inputStream the java.io.Inputstream to read data from
  * @param bucketSize the size of each individual bucket
+ * @param maxNumberOfBuckets the maximum number of buckets that can be created before beginning to remove old ones
  */ 
-class BucketingInputSource(inputStream: java.io.InputStream, bucketSize: Int = 1 << 13)
+class BucketingInputSource(inputStream: java.io.InputStream, bucketSize: Int = 1 << 13, maxNumberOfBuckets: Int = 32768)
```

Comment:
We'd have to change the API in order to support it as a tunable. Right now, the user creates the InputSourceDataInputStream explicitly, which doesn't know anything about tunables. We either need a way to get tunables from the DataProcessor and pass it in as an argument, or maybe the DataProcessor becomes a factory that creates InputSourceDataInputStreams and itself passes in the necessary tunable information?

Reply:
Speaking of which, our current API doesn't allow changing the bucketSize or this new maxNumbeOfBuckets capabilities. The JAPI and SAPI only allow passing in an InputStream, but has not configurability. Perhaps supporting that from tuanbles becomes part of this new API. Probably needs to be part of the next release so that people can grow this limit if needed, since before it was only limited by memory.